### PR TITLE
Enhancement 2121: overwrite computational services limits

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- set sidecar timeout for services to 0 --> no timeout
- allow to set the limits for CPUs and memory per service through the docker image labels (0 should be unlimited)

## Related issue/s

implements #2121 

## How to test

1. 
  ```bash
  make build-x up-prod
  ```
2. run a computational service with memory changes

### CODE REVIEW

**Review by mguidon — COMMENTED** [review]
Very good. So the sidecar takes the default limits from the env and then every service can override via labels. Is that right? Beautiful.

File: packages/models-library/src/models_library/service_settings.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from typing import Any, List
+
+from pydantic import BaseModel, Extra, Field
+
+
+class ServiceSetting(BaseModel):
+    name: str = Field(..., description="The name of the service setting")
```
Comment (GitHK) [inline packages/models-library/src/models_library/service_settings.py:7]: may I suggest an enum here?

**Review by GitHK — APPROVED** [review]
Nice, I will use this for the service-sidecar as well.

**Review by odeimaiz — APPROVED** [review]

File: packages/models-library/src/models_library/service_settings.py (hunk 1 above)
Comment (sanderegg) [reply packages/models-library/src/models_library/service_settings.py:7]: no that is a free text.

**Comment by sanderegg** [comment]
> Very good. So the sidecar takes the default limits from the env and then every service can override via labels. Is that right? Beautiful.

exactly so.

**Pull request merged**